### ISSUE TITLE
Validate new Kubernetes Versions

### ISSUE THREAD

**Issue by adracus** [onmetal/matryoshka#31]
# Summary

We did not yet validate whether the Kubernetes versions
* `1.22`
* `1.23`

work with Matryoshka. For that, we should spin up an `etcd`, `KubeAPIServer` and `KubeControllerManager` to see whether the cluster is still operational as expected. This means e.g. creating namespaces, secrets, rolebindings etc. works.

**Comment by GrigoriyMikhalkin** [comment]
**TLDR**: matryoshka works with both K8s versions `1.22` and `1.23`. Namespaces/secrets/roles/rolebindings were created successfully.

Compatibility of [latest version](https://github.com/onmetal/matryoshka/commit/7d08057896bc583cfd903e6a1ffb1e1cc4f71343) of matryoshka with K8s versions `1.22` and `1.23` was tested. For testing i used cluster created with latest version of [kind](https://kind.sigs.k8s.io/) with following config:
```
apiVersion: kind.x-k8s.io/v1alpha4
kind: Cluster
nodes:
  - role: control-plane
    image: kindest/node:v1.23.0@sha256:49824ab1727c04e56a21a5d8372a402fcd32ea51ac96a2706a12af38934f81ac
  - role: worker
    image: kindest/node:v1.23.0@sha256:49824ab1727c04e56a21a5d8372a402fcd32ea51ac96a2706a12af38934f81ac
```


Here is steps to reproduce testing routine:

1. Create kind cluster: `kind create cluster --config=kind_config.yaml`
2. Install matryoshka CRDs: `make install`
3. Run matryoshka controller: `make run`
4. We will use manifests from `config/samples` to deploy components. Update apiserver and controller-manager to desired version(just search for `version:` in `config/samples` directory).
5. Update `etcd` to desired version(specify image for `apiserver-etcd-sample` Deployment)
6. Apply manifests: `kubectl apply -f config/samples`
7. After all 3 pods are ready, set connection to newly created apiserver. For that create a node port:
```
apiVersion: v1
kind: Service
metadata:
  name: kubeapiserver-apiserver-sample
spec:
  type: NodePort
  selector:
    matryoshka.onmetal.de/app: kubeapiserver-apiserver-sample
  ports:
    - port: 443
      targetPort: 443
      nodePort: 30007
```
8. Test connection to apiserver: ` curl https://workerNodeContainerIP:30007/api --header "Authorization: Bearer some-token" --insecure`
9. Create kubeconfig:
```
apiVersion: v1
clusters:
- cluster:
    server: https://workerNodeContainerIP:30007
  name: kind-kind
contexts:
- context:
    cluster: kind-kind
    user: admin
  name: kind-kind
current-context: kind-kind
kind: Config
users:
- name: admin
  user:
    token: some-token
```
10. Now you can start testing resources creation. To interact with apiserver use command:
```
kubectl --kubeconfig=./kubeconfig --insecure-skip-tls-verify ...
```
